src/lib/package-lib.js

```diff
@@ -0,0 +1,106 @@
+const fs = require('fs');
+const path = require('path');
+const TerserPlugin = require('terser-webpack-plugin');
+const webpack = require('webpack');
+
+const fsUtils = require('../lib/sync-fs');
+const { info, warn, error } = require('./log');
+
+module.exports = (pathToProject, entry, baseEslintPath, options = {}) => {
+  const baseEslintConfig = fsUtils.readJson(baseEslintPath);
+  
+  const directoryContainingEntry = path.dirname(entry);
+  const libName = path.basename(directoryContainingEntry);
+  info(`Packaging ${libName}`);
+
+  const outputFilename = `./${libName}.js`;
+  const outputDirectoryPath = path.join(__dirname, '../../build');
+  createDirectoryIfNecessary(outputDirectoryPath);
+
+  const compiler = webpack([{
+    mode: 'production',
+    entry,
+    output: {
+      pathinfo: false,
+      filename: outputFilename,
+      path: outputDirectoryPath,
+    },
+    optimization: {
+      minimize: !!options.minifyScripts,
+      minimizer: [
+        new TerserPlugin({
+          terserOptions: {
+            warnings: true,
+            output: {
+              comments: false,
+              quote_style: 1,
+            },
+          },
+        }),
+      ],
+    },
+    resolve: {
+      alias: {
+        'tasks.js': path.join(pathToProject, 'tasks.js'),
+        'targets.js': path.join(pathToProject, 'targets.js'),
+        'contact-summary.templated.js': path.join(pathToProject, 'contact-summary.templated.js'),
+      },
+    },
+    module: {
+      rules: [
+        {
+          enforce: 'pre',
+          test: /\.js$/,
+          loader: 'eslint-loader',
```

Comment:
I tried running this over `medic/conf/standard` and kept getting this error:
```
ERROR in Entry module not found: Error: Can't resolve 'eslint-loader' in 
'/home/diana/projects/medic/config/standard'
```

Am I doing something wrong?

Reply:
Standard's package.json references `"medic-conf": "^2.0.4"`.  The `eslint-loader` won't be present until we bump to this newer version.  It is a bit weird that this webpack is looking in the `config/standard/node_modules` instead of its `medic-conf/node_modules` ...  Unsure if that is a bug.

Reply:
Ok, I tried a different exercise. I created a new project with `initialise-project-layout` (which has no package.json) and then immediately `compile-app-settings` and seeing the error again.

Reply:
Right. That is what I'm saying -- this change would require that we add a package.json into `initialise-project-layout`. I also haven't published this update to npm yet.

Reply:
Wouldn't it be simpler if webpack actually got the loader from the right place? :)

Reply:
Yes

Reply:
Well, let's do that, then.

Reply:
Okay. That was actually pretty easy. Good suggestion, thanks.